dev-libs/rocm-opencl-runtime/rocm-opencl-runtime-5.1.3.ebuild

```diff
@@ -0,0 +1,124 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit cmake edo flag-o-matic prefix
+
+DESCRIPTION="Radeon Open Compute OpenCL Compatible Runtime"
+HOMEPAGE="https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime"
+SRC_URI="https://github.com/ROCm-Developer-Tools/ROCclr/archive/rocm-${PV}.tar.gz -> rocclr-${PV}.tar.gz
+	https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime/archive/rocm-${PV}.tar.gz -> rocm-opencl-runtime-${PV}.tar.gz"
+
+LICENSE="Apache-2.0 MIT"
+SLOT="0/$(ver_cut 1-2)"
+IUSE="debug test"
+RESTRICT="!test? ( test )"
+KEYWORDS="~amd64"
+
+RDEPEND=">=dev-libs/rocr-runtime-${PV}
+	>=dev-libs/rocm-comgr-${PV}
+	>=dev-libs/rocm-device-libs-${PV}
+	>=virtual/opencl-3
+	media-libs/mesa"
+DEPEND="${RDEPEND}"
+BDEPEND=">=dev-util/rocm-cmake-${PV}
+	media-libs/glew
+	test? ( >=x11-apps/mesa-progs-8.5.0[X] )
+	"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-5.1.3-remove-clinfo.patch"
+	"${FILESDIR}/${PN}-3.5.0-do-not-install-libopencl.patch"
+)
+
+S="${WORKDIR}/ROCm-OpenCL-Runtime-rocm-${PV}"
+S1="${WORKDIR}/ROCclr-rocm-${PV}"
+
+# CMAKE_BUILD_TYPE=Release
+
+src_prepare() {
+	# Remove "clinfo" - use "dev-util/clinfo" instead
+	[ -d tools/clinfo ] && rm -rf tools/clinfo || die
+
+	cmake_src_prepare
+
+	hprefixify amdocl/CMakeLists.txt
+
+	sed -e "s/DESTINATION lib/DESTINATION ${CMAKE_INSTALL_LIBDIR}/g" -i packaging/CMakeLists.txt || die
+	# remove trailing CR or it won't work
+	sed -e "s/\r$//g" -i tests/ocltst/module/perf/oclperf.exclude || die
+
+	pushd ${S1} || die
+	# Bug #753377
+	# patch re-enables accidentally disabled gfx800 family
+	eapply "${FILESDIR}/${PN}-5.0.2-enable-gfx800.patch"
+	popd
+}
+
+src_configure() {
+	# Reported upstream: https://github.com/RadeonOpenCompute/ROCm-OpenCL-Runtime/issues/120
+	append-cflags -fcommon
+
+	local mycmakeargs=(
+		-Wno-dev
+		-DROCCLR_PATH="${S1}"
```

Comment:
Why can't we use the rocclr installed by `dev-libs/rocclr`?

Reply:
Good question. In ROCm-4.3.0 we are packaging rocclr, but staring from ROCm-5.0, rocm-opencl-runtime and hip are bundling rocclr, which is recommended by ROCclr itself: https://github.com/ROCm-Developer-Tools/ROCclr/blob/cf7a485fbf0b1326a5517f43a67e5bd6936f8699/CMakeLists.txt#L26. So `dev-libs/rocclr` is unmaintained from ROCm-5.
But from what I can see, maintaining  `dev-libs/rocclr` is an option. What do you think?

Reply:
Hmm, it looks like there are a bunch of other roc/hip packages that also bundle rocclr. I think unbundling it here only makes sense if we (can) also unbundle it in those other packages. 

In general I think it is better to unbundle things, but if upstream explicitly does not support it then maybe unbundling will open up a whole new can of worms and that might not be worth the effort.

Lets keep it like this for now, and then maybe we can revisit the rocclr bundling later.